### DISCUSSION TITLE
FNIRSI GC-01: Can't install

### DISCUSSION THREAD

**Discussion by Urminme** [Gissio/radpro#145] [Q&A]
I have tired connecting The GC-01 to my computer and when I plug it in I get a windows message stating there is an issue with the USB device, tried with several cables, I opened Device manager and when I hold power down it does seem to have something change on Device manager as it resets but im just not sure what it is. Do I need to have some specific driver or is there something wrong with my Geiger counter?

**Comment by ChyrkaA** [comment]
No, everything is fine with your device. It needs to be reflashed, most likely through a programmer. Or from Ubuntu. I had the same situation, you can read in this thread:
https://github.com/Gissio/radpro/discussions/133

**Comment by Urminme** [comment]
> No, everything is fine with your device. It needs to be reflashed, most likely through a programmer. Or from Ubuntu. I had the same situation, you can read in this thread: #133

So I was able to get it programmed and had to use the ST-link method unfortunately, but im not sure if I didnt set it right and my counter is all over the place it never dips below 0.500 uSv/h and often goes way up to over 1.000uSv/h which is not possible there is nothing that would give off that much everywhere I go, is there a way to tell if my tube is bad or something

**Comment by dc1rdb** [comment]
Your HV might be too high. I would suggest you go to a custom HV profile in your Geiger settings and start with a real low value, like 5 kHz and 1% PWM duty cycle. Then check if your counter readings stabilize at normal values.

**Comment by Urminme** [comment]
> Your HV might be too high. I would suggest you go to a custom HV profile in your Geiger settings and start with a real low value, like 5 kHz and 1% PWM duty cycle. Then check if your counter readings stabilize at normal values.

Thank you for the help, so I tried 5kHz and 1% but it just did nothing after that no measurements, I went back to the standard and it started working again, its weird the meter is always measuring high and sporadic no matter where I am at or what im doing, sometimes its lower at 0.300 and randomly jumps up to 1.300 then back down to 0.600 its just all over the place, Ive tried messing with background compensation which does not make much difference. It just seems horribly inaccurate and im very doubtful its Radpro as it was very squirrely before I switched over to this much much better firmware, I was hoping this would make things more accurate and stable but im not sure what else to assume other then the tube is just shitty or something else is causing erroneous readings, its using a M4101 tube and set to such in the geiger settings

**Comment by dc1rdb** [comment]
No offense, but have you double checked correct polarity of your geiger tube? (I had that problem once)

**Reply by Urminme** [reply]
So they can be installed backwards from the manufacturer? I will give it a check and see

**Comment by Gissio** [comment]
Converting this issue to a discussion (#32).

**Reply by Gissio** [reply]
Try setting a custom HV profile with these parameters:

PWM Frequency: 10 kHz
PWM Duty Cycle: 75%

If that doesn’t work, try doing what dc1rdb suggested above: Set up a PWM frequency of 5 kHz and a duty cycle of 1 %, and slowly increase the duty cycle until you get consistent counts. With a M4011 tube, the background level should be around 20 counts per minute.

If that doesn’t work either, most likely your device is faulty.

**Reply by ugopapetti** [reply]
Yes, know that's not the correct place for this, however... why don't use one of the 4 unused processor ports configuring them as ADC and measure/display the HV trough a simple HV voltage divider? 
There is enough space for such small hack (talking for the GC-01 i have in my hands). 

Know that if the pwm does not perform as programmed this may add entropy however there will be a "standard" way to measure HV.

**Reply by ugopapetti** [reply]
This could be a feasible HW schematic, necessary to ensure a correct impedence to the microprocessor ADC pin. 

Load on HV could be less however over 400 MOhm should be sufficient to don't perturbate too much the HV generator. Using 8 resistors reduces  the need for specialized HV components.

The R38 resistor present on some GC-01 pcb's (51MOhm) could be removed to reduce total current. HV is pulled from R38 HV side. 

Output connected to an available microprocessor pin. 

PCB could eventually be made on a thin and flexible pcb (need to check the HV implications). 

PCB could be shaped in a way it could be placed on some free space of pcb, minimizing wiring.

R9 is a compromise, allowing a low voltage divider out impedance, minimizing potential noise and allowing for a safe overvoltage protection; could be raised to have additional ADC resolution.

![image](https://github.com/user-attachments/assets/911ae38b-b6a7-4526-b3cd-d675135aba80)

![image](https://github.com/user-attachments/assets/8817dd94-ff31-4c9c-bb1d-69004022a749)

**Comment by ullix** [comment]
A voltage scan can easily be done with my software GeigerLog. The picture shows a recent scan with a FNIRSI GC-01 (yellow) and firmware radpro 2.0.2. The computer does modify the voltage automatically. When the voltage goes up and down between 300 and 800 V, you can see how the countrate creeps up and down the Geiger plateau (the "plateau" is never perfectly flat).

Download GeigerLog pre-release 1.5.1pre15  from https://sourceforge.net/p/geigerlog/discussion/devel2/

![image](https://github.com/user-attachments/assets/c9e4592c-c22e-4a75-8339-843099c11a17)

**Reply by alfmck** [reply]
To ullix: Thank You for Geigerlog 1.5.1pre15. I still used v1.4.3d and have found new useful enhancements!
Would be interesting to know how to use "Enable Radpro Anode Voltage Control" function for GC-01?

**Comment by ullix** [comment]
@alfmck : if you are using pre15 as downloaded, it should already be configured for automatically using Anode Voltage Control! You will get infos printed into NotePad when logging. 
It is not in your case? What device are you using and RadPro firmware?

**Reply by alfmck** [reply]
I'm using yellow GC-01 with Geeky CPU and Radpro 2.0.2. As has written in Geigerlog.cfg radpro configuration section, Anode Voltage Control is used for stabilizing the HV. How are you reading the HV in GC-01? As you've writing before, digital voltmeter, connected to PC and controlled by Geigerlog, could be used. Or ugopapetti suggested HV schematic could be used, but it has to be programmed in Radpro.

**Reply by ullix** [reply]
Hang in for a moment, because I have some bad news to report. But for now:
I had a **yellow** GC-01 with chip **CH32F103R8T6 from WCH**, which I so far had used for all development.
Now I also have a **blue** GC-01 with chip **Geehy APM32F103RBT6**.

So far I thought yellow==WCH, and blue==Geehy,  but since you say to have a yellow-Geehy, there may only be a random relationship between case color and chip used?

**Reply by alfmck** [reply]
![IMG_3445](https://github.com/user-attachments/assets/ff270824-c460-48f3-9221-7510c01bcae9)
![IMG_3444](https://github.com/user-attachments/assets/e2a04fb9-1510-421d-9adc-5d8237adee55)

**Reply by ullix** [reply]
Looks yellow to me :-/

**Reply by ullix** [reply]
Can you please post what GeigerLog shows when you click RadPro Device Info? On my Geehy I see:

Hardware ID:                  FNIRSI GC-01 (APM32F103CB)
Software ID:                  Rad Pro 2.0.2

**Comment by ullix** [comment]
I have now a blue-case GC-01 with a Geehy APM32F103RBT6 chip, and oddly enough, it behaves differently from my yellow-case GC-01 with a CH32F103R8T6 chip from WCH. The Geehy unit performs **more poorly** than the device with WCH chip!

I installed RadPro 2.0.2 on both devices; easy job on both. And judging from the interaction of GeigerLog with either device, the firmware acts the same way on both.

In particular, when I use GeigerLog via RadPro to set PWM (Freq, Duty), I can observe at PA3 of both chips that the PWM signal is indeed what I attempted to set it to at within 1% or better by using an oscilloscope.

Comparing both circuit boards visually, I do not see any difference! Can someone else confirm or point out differences?

However, the Geehy board performs **VERY** differently with respect to anode voltage, and I don't like it.

Next pic shows a scan where GeigerLog used the RadPro firmware to step through the PWM frequency (red curve) at a duty of 50%, reads the counts reported by RadPro (approx CPS=40, orange curve), and also ready the anode voltage using an OW18E DMM via Bluetooth (green curve).

Unsurprisingly, no counts are created when the anode voltage is too low. The max anode voltage reached is 600V at 16 kHz and 50% duty (magenta dotted line). Increasing freq further leads to voltage **de**crease, with an odd saw-tooth pattern, and finally a collapse at near 70kHz. 

Looking at the full curve, the factory default (490V) PWM setting is at an odd place (56.7kHz, 50%). 

From my perpective a workable range is black-circled allowing 350...600V (4...16kHz, 50%). This covers voltages for many tubes, but it is far away from what can be done with a counter with WCH chip, [namely **1200 V** !!! ](https://www.geigerzaehlerforum.de/index.php/topic,2277.msg30457.html#msg30457). [Lateron ](https://www.geigerzaehlerforum.de/index.php/topic,2277.msg30986.html#msg30986) I show that a freq from 100 to 600 Hz, can cover an anode voltage range from 300V to 800V smoothly.

![Selection_004](https://github.com/user-attachments/assets/02cce74d-8825-4f4b-a56a-4c19a03e113f)

So, for setting **490V** we have the options for PWM settings:

- Geehy chip:    
Freq=56700 Hz  Duty=50% (Factory Default)
Freq=  7000 Hz  Duty=50% (my suggested setting)
- WCH chip:      
Freq=     240 Hz  Duty=50% (my suggested setting)

The settings are grossly different despite: Firmware is the same version, PWM signal from the chip is the same, Circuit board (looks like | is) the same. How comes?

For completeness I did a quick scan of the duty from 0 to 100% at a freq=15 kHz; result shown in next pic. Max voltage reached remains at 600V.

![image](https://github.com/user-attachments/assets/f154a0bd-a50c-4b0a-800b-bd5f955ef29a)

**Reply by alfmck** [reply]
In my GC-01 R38 is not soldered, so I get higher HV:
7000$5.5% - 580V 
7000&10% - 720V
7000&20% - 850V
7000&50% - 920V
I use two big sensitive tubes, connected parallelly, so I get ambient about 250-300 CPM. I noticed, that in order to get more sensitivity and accuracy, you have to use higher HV value on plato.

**Reply by ullix** [reply]
Yes, higher voltage gives higher sensitivity. And then even a little bit more voltage, and voila, you have a nice desk lamp!  [https://youtu.be/avdHJTEzvmQ](https://youtu.be/avdHJTEzvmQ)

**Comment by ullix** [comment]
Is somebody able to explain what this R38 does?

In my 2 devices I found these implementations:

```
WCH chip     yellow case     R38= '516' = 51 MOhm    voltage up to 1200 V
GEEHY chip   blue case       R38='5105' = 51 MOhm    voltage up to 600 V
```
No difference. This cannot be the cause for the big difference in max voltage.

**Reply by alfmck** [reply]
I guess that is some fault in the blue case. Because we have schematics, and you have 2 devices, try to compare voltages in HV circuits with oscilloscope. My device also uses Geehy chip, so I could test HV circuits in it too.

**Reply by ullix** [reply]
All numbers I have given on Volt, Freq, and Duty are derived from Osci and/or DMM, and are consistent with the PWM settings done with GeigerLog via RadPro2.0.2!

You can try the latest GeigerLog pre16 - it should cover devices with either WCH and GEEHY chip. 

Your device with modified R38 might behave differently. The please use the RadPro Scan function from the GeigerLog menu to establish the Volt vs PWM curves.

**Reply by alfmck** [reply]
I'm unable to use Radpro "Anode Voltage Control" function in pre16 version, because base frequence 5573Hz for HV=450V is not acceptable - it sets HV about 800V in my device.
I use HV about 560V when set 64400&50% or 300&50%. Values 300&50% are used in version pre15 and fit for me, but as I noticed, pre16 is more advanced.
I didn't find how to use RadPro Scan function from the GeigerLog menu. Good solution would be to transfer data from pre15 in pre16.
And one more asking: Could You extend Cycle set button max value from 1000s to 3600s. I hope, that if I set cycle=3600s, I get AVERAGE CPM value for 1 hour. In this case I will get average ambient radiation values for every 1 hour, like in https://remap.jrc.ec.europa.eu/Advanced.aspx
and graph format in EU site and Geigerlog are the same! In this case instead of using Datalog in Radpro I could use Quick log function in geigerlog and get a real time graph data!

**Reply by ullix** [reply]
@alfmck : I am happy to address all these questions, but if this goes deeper into GeigerLog, I suggest to do it on the GeigerLog discussion site (https://sourceforge.net/p/geigerlog/discussion/) . Just create a topic. 

We haven't heard back from the original poster @Urminme  whether this anode voltage issue might explain his problems?

This is the crux with custom versions! We already have the WCH type and the Geehy type for Anode voltage setting. Your version is the third variant. And presently I am not even sure that all WCH and Geehy types will behave the same? I could make the Volt vs Freq curve configurable by the user, but would the user be able to create it? Perhaps just a rough version like frequency for voltages from 300...800V in steps of approx. 100V?

Databses from any GeigerLog version can be read in every other GeigerLog version. Just load the respective * .logdb file.

The GeigerLog database holds 2, and only 2 count values: CPM and CPS. There is no CPH (Counts per hour). If you were extending the logcycle to 3600 sec, you would get these two values for the last sec, or min, resp., not an average!

However, you could use a formula in the formula interpreter, like: `CPM1st = COUNTRATE_AVG("CPM", "hour")` which would record the hourly average of variable CPM in variable CPM1st.

**Reply by alfmck** [reply]
The new version of geigerlog is a great instrument for accurate setting a PWM to HV relation. The problem is that after disconnecting the multimeter, the HV value increases for about 200V. So, in order to keep previous value of HV you have to correct PWM or permanently connect 1GOm resistor parallelly to the tube. I tried connecting multimeter parallelly to R38 which acts as HV stabilizer. But in my GC-01 R38 is not soldered and HV increases about 50V too. In your case connecting multimeter parallelly to R38 when using Scan function would be a good solution.
50% duty cycle works for me in pre15, where low frequences are used. But in pre16 you set about 5khz, while for correct HV on tube I need 64.4khz. So even choosing HV=300V, I get unacceptable 700V on my tube.
It's a pity that geiherlog not hold CPH value, which is widely used in monitoring stations. The formula you give does not work for me.
And once more: In your ARM device max voltage is 600V. It is about twice less than in CH device. It could be if one of HV multiplier stage does not work. I measured voltages on C14, C15, C16 with multimeter and 1GOm resistor and got clean about 80V, 160V, 320V, 640V values.

**Reply by ullix** [reply]
I have presented data that show that the two GC-01 devices, the one based on WCH chip, and the one based on GEEHY chip,  are giving very different result with respect to HV generation. What makes you expect that your device, different from both of these, would give the same result than either one? IT DOES NOT! So, do scans for both Freq and Duty, and measure anode voltage via a 1GOhm resistor, and report the result. ONLY THEN can a function be defined, which represents your modified device.

And do measure the anode voltage across the tube, from cathode to anode, and not elsewhere.

**Reply by alfmck** [reply]
to ullix: I've made measurements and changes to Geigerlog pre16 with Duty cycle 2.25%:
![2024-11-14_164737](https://github.com/user-attachments/assets/8d7004b3-24d3-4d28-ac4f-6b0a33dd26e5)

![2024-11-14_164456](https://github.com/user-attachments/assets/80d6a36b-67b1-4833-b030-6d9498cf2401)
Thank You very much for help! I'm happy using Geigerlog. Especially Notepad and professional graph windows!

**Comment by ullix** [comment]
Good to see it is working!

However, summarizing the 3 available control curves for a GC-01, we now got this:

![Screenshot at 2024-11-15 12-47-04](https://github.com/user-attachments/assets/ca52d2dd-0e4c-419a-82af-40daa08e682b)


Strange, to say the least. And then there are all these other devices which do run RadPro ...

The curves for GC-01 devices WCH and GEEHY were generated with Duty=50%. Yours were done with 2.25%. Can you do a Freq scan also with 50%?

**Reply by alfmck** [reply]
For duty=50% graphs are two:
![2024-11-15_172848](https://github.com/user-attachments/assets/d7997dc7-de6e-49fd-9a12-b4e88e92a17c)
![2024-11-15_173326](https://github.com/user-attachments/assets/124471c1-3d2d-46ae-beb1-200a92cb6ef9)

**Comment by ullix** [comment]
I have added your new data to the graph. 
Note that the blue WCH curve is on the right scale, all others are on the left!

![Screenshot at 2024-11-16 09-28-10](https://github.com/user-attachments/assets/af7e2e49-1db8-4dae-86f3-a1754ade8901)

The new high-frequency curve (brown) is too flat for being a good control curve; the low freq curve (green) is ok.

The difference between blue, red, green is amazing - it will be hard to define some generic formula!

I'm beginning to wonder whether it has anything to do with the RadPro version ....

**Reply by ugopapetti** [reply]
Hi Ullix,
see just now your entry above..

L1 is different between models / production batches. Please have a look here:

![image](https://github.com/user-attachments/assets/552f8969-566f-4dcf-96a9-6bfbfed3d278)

You can observe that L1 on left picture (WCH processor) is marked "103" that corresponds to 10mH while the L1 on the right (GEEHY) is marked "102", 1mH.

This will cause the switching circuit (L1 / Q3) to operate at lower frequency with a 10mH inductor and at an higher frequency with a 1mH inductor, who is clearly evident in your picture above to produce the same order of magnitude HV, (150Hz to 1Khz frequency for circuit equipped with 10mH inductor compared to 2KHz to 20KHZ for 1mH). In my GEEHY GC-01, L1 is 1mH (tube is a M4011).

This explains the difference in behavior, that is not due to the processor or to the RadPro, instead to the inductor value installed in the specific production batch (between 1mH and 10mH there is an order of magnitude of difference).

ugo

**Comment by ullix** [comment]
@ugopapetti : Now that you said it ...
Yes, I confirm. Many thanks for pointing this out.

Just a minute ago I posted on the difference between these two GC-01 versions here: 
https://sourceforge.net/p/geigerlog/discussion/general/thread/beafedc7b1/?page=11&limit=25#d560

A third order parabola does an excellent fit for the Freq vs Anode Voltage function, but a much simpler linear approximations for the two devices suffices:
```
# WCH   Freq = -317.174 + 1.52174 * AVoltage
# GEEHY Freq = -10463.2 + 37.8947 * AVoltage
```
Makes for some frustration in programming. Is there any electrical reason for preferring one over the other?

**Reply by dc1rdb** [reply]
If you guys are collecting data on the different build standards of GC-01 units, here is mine:
Original Fnirsi, Geehy APM32 processor, L1 102, R38 not fitted, J321 tube.

**Reply by ullix** [reply]
Yes, so we now have 3 observations of the GC-01 with GEEHY chip has 1mH inductivity, and 2 observations that with WCH chip there is 10mH! Gives hope for consistency ;-)

But what about this R38? Is it missing on the factory-fresh circuit board? Or did you remove it? Or is it shortened? 

On my GEEHY device I read R38 as 5105 (51 MOhm) and on the WCH 516 (also 51MOhm).

And whatever the change: what is the impact by it?

**Reply by dc1rdb** [reply]
R38 was not fitted by the factory on my unit. The solder pads are open.

**Reply by ullix** [reply]
What does R38 do?

**Reply by dc1rdb** [reply]
Refer to the schematics: https://github.com/Gissio/radpro/blob/main/docs/devices/FNIRSI%20GC-01/schematics.pdf
In my understanding, it acts as a load for the high voltage. I will probably put a 400 V zener there instead.

**Comment by alfmck** [comment]
Does somebody know, how HV circuits are constructed in FS-5000? According to https://github.com/Gissio/radpro/blob/main/docs/devices/Bosean%20FS-5000/peripherals.md
PB6, PB7 are used for HV setting, but Gissio didn't make a HV settings possibility in settings menu. Maybe Zener is used in it?
